src/constructors.cpp

```diff
@@ -47,14 +47,28 @@ namespace manifold {
  * can be sharpened by sharping all edges that are incident on it, allowing
  * cones to be formed.
  */
-Manifold Manifold::Smooth(const MeshGL& meshGL,
+Manifold Manifold::Smooth(MeshGL meshGL,
                           const std::vector<Smoothness>& sharpenedEdges) {
   DEBUG_ASSERT(meshGL.halfedgeTangent.empty(), std::runtime_error,
                "when supplying tangents, the normal constructor should be used "
                "rather than Smooth().");
 
+  const std::vector<uint32_t> faceIDin = meshGL.faceID;
+  meshGL.faceID.resize(meshGL.NumTri());
+  std::iota(meshGL.faceID.begin(), meshGL.faceID.end(), 0);
+
   std::shared_ptr<Impl> impl = std::make_shared<Impl>(meshGL);
   impl->CreateTangents(impl->UpdateSharpenedEdges(sharpenedEdges));
+  // Restore the original faceID
```

Comment:
Maybe move this out into a function.

Reply:
Yeah, not sure why I didn't do that before...